itest/client.go

```diff
@@ -0,0 +1,240 @@
+var (
+	ErrTimeout = fmt.Errorf("Transaction be on chain timeout")
+)
+
+type Client struct {
+	grpc  rpc.ApisClient
+	mutex sync.Mutex
+	Name  string
+	Addr  string
+}
+
+func (c *Client) getGRPC() (rpc.ApisClient, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.grpc == nil {
+		conn, err := grpc.Dial(c.Addr, grpc.WithInsecure())
+		if err != nil {
+			return nil, err
+		}
+		c.grpc = rpc.NewApisClient(conn)
+		ilog.Infof("Create grpc connection with %v successful", c.Addr)
+	}
+	return c.grpc, nil
+}
+
+func (c *Client) GetTransaction(hash string) (*Transaction, error) {
+	grpc, err := c.getGRPC()
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := grpc.GetTxByHash(
+		context.Background(),
+		&rpc.HashReq{
+			Hash: hash,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	transaction := &Transaction{
+		Tx: (&tx.Tx{}).FromPb(resp.Tx),
+	}
+
+	return transaction, nil
+}
+
+func (c *Client) GetReceipt(hash string) (*Receipt, error) {
+	grpc, err := c.getGRPC()
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := grpc.GetTxReceiptByTxHash(
+		context.Background(),
+		&rpc.HashReq{
+			Hash: hash,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt := &Receipt{
+		TxReceipt: (&tx.TxReceipt{}).FromPb(resp.GetTxReceipt()),
+	}
+
+	return receipt, nil
+}
+
+func (c *Client) GetAccount(name string) (*Account, error) {
+	grpc, err := c.getGRPC()
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := grpc.GetAccountInfo(
+		context.Background(),
+		&rpc.GetAccountReq{
+			ID:              name,
+			UseLongestChain: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Get account permission by resp
+	account := &Account{
+		ID:      name,
+		Balance: resp.GetBalance(),
+	}
+
+	return account, nil
+}
+
+func (c *Client) SendTransaction(transaction *Transaction) (string, error) {
+	grpc, err := c.getGRPC()
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := grpc.SendTx(
+		context.Background(),
+		&rpc.TxReq{
+			Tx: transaction.ToPb(),
+		},
+	)
+	if err != nil {
+		return "", err
+	}
+
+	if err := c.checkTransaction(resp.GetHash()); err != nil {
+		return "", err
+	}
+
+	return resp.GetHash(), nil
+}
+
+func (c *Client) checkTransaction(hash string) error {
+	ticker := time.NewTicker(Interval)
+	afterTimeout := time.After(Timeout)
+	for {
+		select {
+		case <-afterTimeout:
+			return ErrTimeout
+		default:
+			<-ticker.C
+			r, err := c.GetReceipt(hash)
+			if err == nil && r != nil {
+				if r.Success() {
+					return nil
+				} else {
+					return fmt.Errorf("%v: %v", r.Status.Code, r.Status.Message)
+				}
+			}
+		}
+	}
+}
+
+func (c *Client) CreateAccount(creater *Account, name string, key *Key) (*Account, error) {
+	action1 := tx.NewAction(
+		"iost.auth",
+		"SignUp",
+		fmt.Sprintf(`["%v", "%v", "%v"]`, name, key.ID, key.ID),
+	)
+
+	action2 := tx.NewAction(
+		"iost.token",
+		"transfer",
+		fmt.Sprintf(`["%v", "%v", %v, %v]`, InitToken, creater.ID, name, InitAmount),
+	)
```

Comment:
You'd better pledge gas and buy ram for the new account.
Or else the new account have no gas, so it can do nothing.